packages/flutter/lib/src/cupertino/page_scaffold.dart

```diff
@@ -32,41 +31,52 @@ class CupertinoPageScaffold extends StatelessWidget {
   ///
   /// If translucent, the main content may slide behind it.
   /// Otherwise, the main content's top margin will be offset by its height.
+  ///
+  /// The scaffold assumes the nav bar will consume the [MediaQuery] top padding.
   // TODO(xster): document its page transition animation when ready
-  final PreferredSizeWidget navigationBar;
+  final ObstructingPreferredSizeWidget navigationBar;
```

Comment:
This is a breaking change, please make sure to announce changes like this on flutter-dev.